packages/cli-kit/src/bugsnag.ts

```diff
@@ -0,0 +1,52 @@
+import * as path from './path.js'
+import {Interfaces} from '@oclif/core'
+import Bugsnag from '@bugsnag/js'
+import {realpath} from 'fs/promises'
+
+/**
+ * If the given file path comes from within a plugin, return the relative path, plus the plugin name.
+ *
+ * This gives us very consistent paths for errors thrown from plugin code.
+ *
+ */
+export function cleanStackFrameFilePath(
+  currentFilePath: string,
+  projectRoot: string,
+  pluginLocations: {name: string; pluginPath: string}[],
+): string {
+  const fullLocation = path.isAbsolute(currentFilePath) ? currentFilePath : path.join(projectRoot, currentFilePath)
+
+  const matchingPluginPath = pluginLocations.filter(({pluginPath}) => fullLocation.indexOf(pluginPath) === 0)[0]
+
+  if (matchingPluginPath !== undefined) {
+    return path.join(matchingPluginPath.name, path.relative(matchingPluginPath.pluginPath, fullLocation))
+  }
+  return currentFilePath
+}
+
+/**
+ * Register a Bugsnag error listener to clean up stack traces for errors within plugin code.
+ *
+ */
+export async function registerCleanBugsnagErrorsFromWithinPlugins(plugins: Interfaces.Plugin[]) {
+  // Bugsnag have their own plug-ins that use this private field
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const bugsnagConfigProjectRoot: string | undefined = (Bugsnag as any)?._client?._config?.projectRoot
```

Comment:
This line is a good candidate candidate to break considering we are accessing private properties and we can't rely on Typescript to warn us if Bugsnag has changed the internals of the library. According to the [docs](https://docs.bugsnag.com/platforms/javascript/configuration-options/#projectroot) the default value is `process.cwd()` so we can read that value directly from here.